src/rewriter.coffee

```diff
@@ -316,7 +314,8 @@ exports.Rewriter = class Rewriter
           # Close implicit objects such as:
           # return a: 1, b: 2 unless true
           else if inImplicitObject() and not @insideForDeclaration and sameLine and
-                  tag isnt 'TERMINATOR' and prevTag isnt ':'
+                  tag isnt 'TERMINATOR' and prevTag isnt ':' and
+                  not (tag is 'POST_IF' and startsLine and implicitObjectContinues(i + 1))
```

Comment:
the actual fix. If we see a `POST_IF` on the first line of an implicit object, check if it looks like that implicit object is actually multi-line rather than assuming it's an inline-implicit-object-ending `POST_IF`